src/bokeh/plotting/_plot.py

```diff
@@ -149,6 +150,8 @@ def _get_axis_class(axis_type: AxisType | None, range_input: Range, dim: Dim) ->
         return LogAxis, {}
     elif axis_type == "datetime":
         return DatetimeAxis, {}
+    elif axis_type == "timedelta":
+        return TimedeltaAxis, {}
```

Comment:
might be nice to replace this `elif` with a `match`, as long as we are touching this code. Also I think there are some basic unit tests for `get_scale` that could be expanded: 

https://github.com/bokeh/bokeh/blob/aec1ae64c8d384b325d6cc4988ca2dc95635698a/tests/unit/bokeh/plotting/test__plot.py#L60-L77